src/ripple/app/misc/impl/AmendmentTable.cpp

```diff
@@ -317,23 +319,29 @@ AmendmentTableImpl::AmendmentTableImpl(
     {
         if (auto s = add(a.first, sl))
```

Comment:
I think because it returns a ptr, the original author was being redundant or future-proofing. Instead I rewrote `add()` to return a reference, and changed the callers as appropriate. Now that it's a reference `s` can't be `const` because the name and flags are set.

I do know that you're not a huge fan of `auto`, but I love it, because I'm lazy, and I use an IDE that helps me be lazy (by offering autocomplete, for example), but I'll change it here.